test/spec/Feature/Query/RpcSpec.hs

```diff
@@ -1244,6 +1244,20 @@ spec actualPgVersion =
             let respBody = simpleBody r
             respBody `shouldBe` file
 
+        it "should call the function with no parameters and not fallback to the single unnamed parameter function when using GET with Content-Type headers" $ do
+          request methodGet "/rpc/overloaded_unnamed_param" [("Content-Type", "application/json")] ""
+            `shouldRespondWith`
+              [json| 1|]
+              { matchStatus  = 200 }
```

Comment:
Hm, did `Content-Type: application/json` also failed?

https://github.com/PostgREST/postgrest/issues/2147 only mentions `text/plain` and `octet-stream`.